C/c4Document.cc

```diff
@@ -57,9 +58,14 @@ struct C4DocumentInternal : public C4Document {
 
     void init() {
         docID = _versionedDoc.docID();
-        flags = (C4DocumentFlags)_versionedDoc.flags();
-        if (_versionedDoc.exists())
-            flags = (C4DocumentFlags)(flags | kExists);
+        if(c4doc_isExpired(_db, docID)) {
```

Comment:
This adds significant overhead to loading a document, since another key has to be looked up in the db. I don't think it's worth it to do it for every document load. Instead we could expose c4doc_isExpired in the API. (Or maybe c4doc_getExpiration to return the exact expiration time, with a 0 value meaning "never"?)